scrypto/src/math/integer/convert.rs

```diff
@@ -0,0 +1,788 @@
+                    }
+            }
+            )*
+        }
+    };
+}
+impl_from_primitive! { I8, I16, I32, I64, I128, I256, I384, I512, U8, U16, U32, U64, U128, U256, U384, U512 }
+
+macro_rules! try_from{
+    ($t:ident, ($($o:ident),*)) => {
+        $(
+            paste! {
+                impl TryFrom<$o> for $t {
+                    type Error = ParseIntError;
+                    fn try_from(val: $o) -> Result<$t, ParseIntError> {
+                        BigInt::from(val).try_into().map_err(|_| ParseIntError::Overflow)
+                    }
+                }
+                impl TFrom<$o> for $t {
+                    type Output = $t;
+                    fn tfrom(val: $o) -> Self::Output {
+                        Self::Output::try_from(val).unwrap()
+                    }
+                }
+            }
+        )*
+    };
+}
+
+macro_rules! try_from_large_all {
+    ($($t:ident),*) => {
+        $(
+            try_from! { $t, (I256, I384, I512) }
+            try_from! { $t, (U256, U384, U512) }
+        )*
+    };
+}
+
+try_from_large_all! { U8, U16, U32, U64, U128, I8, I16, I32, I64, I128 }
+
+try_from! {u8, (U16, U32, U64, U128, U256, U384, U512, I8, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {u16, (U32, U64, U128, U256, U384, U512, I8, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {u32, (U64, U128, U256, U384, U512, I8, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {u64, (U128, U256, U384, U512, I8, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {usize, (U128, U256, U384, U512, I8, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {u128, (U256, U384, U512, I8, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {i8, (U8, U16, U32, U64, U128, U256, U384, U512, I16, I32, I64, I128, I256, I384, I512)}
+try_from! {i16, ( U16, U32, U64, U128, U256, U384, U512, I32, I64, I128, I256, I384, I512)}
+try_from! {i32, (U32, U64, U128, U256, U384, U512, I64, I128, I256, I384, I512)}
+try_from! {i64, (U64, U128, U256, U384, U512, I128, I256, I384, I512)}
+try_from! {isize, (U64, U128, U256, U384, U512, I128, I256, I384, I512)}
+try_from! {i128, (U128, U256, U384, U512, I256, I384, I512)}
+
+try_from! {U8, (i8, i16, i32, i64, isize, i128, u16, u32, u64, usize, u128)}
+try_from! {U16, (i8, i16, i32, i64, isize, i128, u32, u64, usize, u128)}
+try_from! {U32, (i8, i16, i32, i64, isize, i128, u64, usize, u128)}
+try_from! {U64, (i8, i16, i32, i64, isize, i128, u128)}
+try_from! {U128, (i8, i16, i32, i64, isize, i128)}
+try_from! {U256, (i8, i16, i32, i64, isize, i128)}
+try_from! {U384, (i8, i16, i32, i64, isize, i128)}
+try_from! {U512, (i8, i16, i32, i64, isize, i128)}
+try_from! {I8, (u8, u16, u32, u64, usize, u128, i16, i32, i64, isize, i128)}
+try_from! {I16, (i32, i64, isize, i128, u16, u32, u64, usize, u128)}
+try_from! {I32, (i64, isize, i128, u32, u64, usize, u128)}
+try_from! {I64, (i128, u64, usize, u128)}
+try_from! {I128, (u128)}
+
+try_from! {U256, (I8, I16, I32, I64, I128, I256, I384, I512)}
```

Comment:
Can we group sort the `try_from!` in a sensible order? A bit hard to read the code.

Reply:
Sorry about that. I will correct those.

Reply:
Fixed in [c5c2c5] https://github.com/radixdlt/radixdlt-scrypto/pull/388/commits/c5c2c5bdb2967b5e8c313e7a251362d649029091